topydo/lib/Config.py

```diff
@@ -311,6 +343,32 @@ def list_format(self):
         """ Returns the list format used by `ls` """
         return self.cp.get('ls', 'list_format')
 
+    def column_keymap(self):
+        """ Returns keymap and keystates used in column mode """
+        keystates = []
+
+        shortcuts = self.cp.items('column_keymap')
+        keymap_dict = dict(shortcuts)
+        for shortcut in shortcuts:
+            combo, action = shortcut
+            combo_list = re.split('(<[A-Z].+?>|.)', combo)[1::2]
+
+            combo_len = len(combo_list)
+            if combo_len > 1:
+                n = 1
+                while n < combo_len:
+                    keystates.append(''.join(combo_list[:n]))
+                    n += 1
```

Comment:
Nice, but `accumulate` should operate on `combo_list` rather than `combo` to avoid creating of garbage keystates when special keys (like `<Esc>` or `<C-s>`) are used.

I would suggest:

```python
import re

from itertools import accumulate

keystates = set()

shortcuts = self.cp.items('column_keymap')
keymap_dict = dict(shortcuts)

for combo, action in shortcuts:
    # add all possible prefixes to keystates
    combo_as_list = re.split('(<[A-Z].+?>|.)', combo)[1::2]
    if len(combo_as_list) > 1:
        keystates |= set(accumulate(combo_as_list[:-1]))
```